catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/web/metrics/IcebergMetricsManager.java

```diff
@@ -0,0 +1,184 @@
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import org.apache.iceberg.metrics.MetricsReport;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IcebergMetricsManager {
+  private static final Logger LOG = LoggerFactory.getLogger(IcebergTableOps.class);
+  public static final String ICEBERG_METRICS_STORAGE = "metricsStorage";
+  public static final String ICEBERG_METRICS_STORAGE_RETAIN_DAYS = "metricsStorageRetainDays";
+  public static final String ICEBERG_METRICS_QUEUE_CAPACITY = "metricsQueueCapacity";
+
+  // Register IcebergMetricsStorage's short name to its full qualified class name in the map. So
+  // that user doesn't need to specify the full qualified class name when creating an
+  // IcebergMetricsStorage.
+  private static final ImmutableMap<String, String> ICEBERG_METRICS_STORAGES =
+      ImmutableMap.of(
+          MemoryMetricsStorage.ICEBERG_METRICS_STORAGE_MEMORY_NAME,
+          MemoryMetricsStorage.class.getCanonicalName());
+
+  private final IcebergMetricsFormatter icebergMetricsFormatter;
+  private final IcebergMetricsStorage storage;
+  private final int retainDays;
+
+  private BlockingQueue<MetricsReport> queue;
+  private Thread metricsWriterThread;
+  private volatile boolean isClosed = false;
+  private ScheduledExecutorService scheduledThreadPoolExecutor;
+
+  public IcebergMetricsManager(IcebergConfig icebergConfig) {
+    icebergMetricsFormatter = new IcebergMetricsFormatter();
+    String metricsStorage = icebergConfig.get(IcebergConfig.ICEBERG_METRICS_STORAGE);
+    storage = loadIcebergMetricsStorage(metricsStorage);
+    LOG.info("Load Iceberg metrics storage: {}", storage.shortName());
+
+    retainDays = icebergConfig.get(IcebergConfig.ICEBERG_METRICS_STORAGE_RETAIN_DAYS);
+    if (retainDays > 0) {
+      scheduledThreadPoolExecutor =
+          new ScheduledThreadPoolExecutor(
+              1,
+              new ThreadFactoryBuilder()
+                  .setDaemon(true)
+                  .setNameFormat("Iceberg-metrics-cleaner")
+                  .setUncaughtExceptionHandler(
+                      (t, e) -> LOG.error("Uncaught exception in thread {}", t, e))
+                  .build());
+    }
+
+    int queueCapacity = icebergConfig.get(IcebergConfig.ICEBERG_METRICS_QUEUE_CAPACITY);
+    Preconditions.checkArgument(
+        queueCapacity > 0,
+        String.format("Iceberg metrics queue capacity:%d should greater than 0", queueCapacity));
+    queue = new LinkedBlockingQueue(queueCapacity);
+    metricsWriterThread =
+        new Thread(
+            () -> {
+              while (Thread.currentThread().isInterrupted() == false) {
+                MetricsReport metricsReport;
+                try {
+                  metricsReport = queue.take();
+                } catch (InterruptedException e) {
+                  LOG.warn("Iceberg Metrics writer thread is interrupted");
+                  return;
```

Comment:
You should not return here. when you catch this exception, you should do some cleanup work.

Also your logic here is not correct, if the thread is interrupted, you will never run the code below.

Reply:
fixed